AirCasting/Extensions/Array+byRemovingDuplicates.swift

```diff
@@ -0,0 +1,50 @@
+// Created by Lunar on 30/08/2022.
+//
+import Foundation
+
+extension Numeric {
+    func decrement() -> Self {
+        self - 1
+    }
+}
+
+extension Array where Element: Equatable & Numeric {
+    /// Iterates all elements and gets rid of the duplicates using provided strategy
+    /// - Parameter strategy: a type of strategy which will be used inside `implement` function.
+    /// If this func returns `nil`, the new element won't be added to the resulting array.
+    /// If it returns a value, this value will be added to the resulting array.
+    /// - Returns: A new `Array` with conflicts resolved.
+    
+    enum StrategyType {
+        case decrementation
+        case none
+    }
+    
+    func byRemovingDuplicates(strategy: Self.StrategyType) -> Self {
```

Comment:
I don't like the name "byRemovingDuplicates". What does it mean?  Maybe it could be more `eliminateDuplicates(by strategy: Self.StrategyType)`? And the call would be: `eliminateDuplicates(by: .incrementation)`. What do you think?

Reply:
I agree, the name is not great. But wouldn't use `by` as a label for parameter. Maybe `using`?